march_rqt_gait_selection/src/march_rqt_gait_selection/gait_selection_view.py

```diff
@@ -0,0 +1,259 @@
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import \
+    QComboBox, QDialog, QGridLayout, QLabel, QPlainTextEdit, QPushButton, QScrollArea, QSizePolicy, QWidget
+from python_qt_binding import loadUi
+import rospy
+
+AMOUNT_OF_AVAILABLE_SUBGAITS = 5
+
+
+class GaitSelectionView(QWidget):
+    def __init__(self, ui_file, controller):
+        """Base class to load and use the gait_selection.ui.
+
+        :param ui_file:
+            Path to the .ui file to load with the widget
+        :param controller:
+            A refrence to the controller corresponding to the rqt gait selection functionalities
+        """
+        super(GaitSelectionView, self).__init__(flags=Qt.Window)
+
+        self._controller = controller
+
+        # Extend the widget with all attributes and children from UI file
+        loadUi(ui_file, self)
+
+        # Coding flags
+        self._colors = {'info': '#000000', 'success': '#009100', 'error': '#FF0000', 'warning': '#b27300'}
+        self._is_refresh_active = True
+        self._is_update_active = True
+
+        # Search for children in GUI
+        self._gait_content = self.Gaits.findChild(QScrollArea, 'GaitSelectionInterface')
+        self._gait_menu = self._gait_content.findChild(QComboBox, 'GaitMenu')
+        self._logger = self._gait_content.findChild(QPlainTextEdit, 'Log')
```

Comment:
As far as I know the `findChild` approach is not necessary for finding widgets. I think you can do just `self.<name>` and then you have it. You should also rename the widgets to lower_snake_case and then you can leave these initializations out and just do `self.gait_content`, `self.logger`, etc.

Reply:
I had some trouble in the beginning with this method but did not try it at the end. I'll have a look to see what is possible.

Reply:
indeed worked, lit